service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/sagemakerNotebook/CreateAwsSagemakerNotebookStep.java

```diff
@@ -0,0 +1,85 @@
+package bio.terra.workspace.service.resource.controlled.cloud.aws.sagemakerNotebook;
+
+import bio.terra.common.iam.SamUser;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.AwsUtils;
+import bio.terra.workspace.generated.model.ApiAwsSagemakerNotebookCreationParameters;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.workspace.AwsCloudContextService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.AwsCloudContext;
+import java.util.Collection;
+import java.util.HashSet;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.services.sts.model.Tag;
+
+public class CreateAwsSagemakerNotebookStep implements Step {
+
+  private final ControlledAwsSagemakerNotebookResource resource;
+  private final AwsCloudContextService awsCloudContextService;
+  private final AuthenticatedUserRequest userRequest;
+  private final SamService samService;
+
+  public CreateAwsSagemakerNotebookStep(
+      ControlledAwsSagemakerNotebookResource resource,
+      AwsCloudContextService awsCloudContextService,
+      AuthenticatedUserRequest userRequest,
+      SamService samService) {
+    this.resource = resource;
+    this.awsCloudContextService = awsCloudContextService;
+    this.userRequest = userRequest;
+    this.samService = samService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    AwsCloudContext cloudContext =
+        awsCloudContextService.getRequiredAwsCloudContext(resource.getWorkspaceId());
+
+    AwsCredentialsProvider credentialsProvider =
+        AwsUtils.createWsmCredentialProvider(
+            awsCloudContextService.getRequiredAuthentication(),
+            awsCloudContextService.discoverEnvironment());
+
+    SamUser samUser = samService.getSamUser(userRequest);
+    Collection<Tag> tags = new HashSet<>();
+    AwsUtils.appendUserTags(tags, samUser);
+    AwsUtils.appendResourceTags(tags, cloudContext);
+
+    FlightMap inputParameters = flightContext.getInputParameters();
+    // TODO(TERRA-550): creationParameters may be used later
+    ApiAwsSagemakerNotebookCreationParameters creationParameters =
+        inputParameters.get(
+            ControlledResourceKeys.CREATE_NOTEBOOK_PARAMETERS,
+            ApiAwsSagemakerNotebookCreationParameters.class);
+
```

Comment:
can you call validateRequiredEntries on these three input param first?
https://github.com/DataBiosphere/terra-workspace-manager/blob/465125a5c64d046279c82095e0fa0cac6ca1c2c6/service/src/main/java/bio/terra/workspace/common/utils/FlightUtils.java#L95